### PR TITLE
Use Gravatar SDK dependency

### PR DESCRIPTION
**Pull request by maxme**
- Add `Gravatar-SDK-Android` as a dependency (latest version `0.2.0`)
- Replace `GravatarUtils` calls from wputils.
- Also bump to latest kotlin version (1.9.10)

cc @hamorillo

### CODE REVIEW

File: WordPressLoginFlow/build.gradle [hunk 1]
```diff
@@ -20,14 +20,18 @@ repositories {
...
+}
+
 android {
     namespace "org.wordpress.android.login"
 
-    compileSdkVersion rootProject.compileSdkVersion
+    compileSdk 34
 
     defaultConfig {
-        minSdkVersion rootProject.minSdkVersion
```
Comment (maxme) [inline WordPressLoginFlow/build.gradle]: I removed references to rootProject.version(s) as this fails when added from WordPress-Android

File: WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupConfirmationFragment.kt [hunk 2]
```diff
@@ -76,6 +76,7 @@ class SignupConfirmationFragment : Fragment() {
             mPhotoUrl = it.getString(ARG_SOCIAL_PHOTO_URL)
             mService = it.getString(ARG_SOCIAL_SERVICE)
         }
+        @Suppress("DEPRECATION")
         setHasOptionsMenu(true)
```
Comment (maxme) [inline WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupConfirmationFragment.kt]: Build fails on wpandroid with deprecations, so I suppressed this one as I'm not planning to rewrite the option menu

File: WordPressLoginFlow/src/main/java/org/wordpress/android/login/util/AvatarHelper.kt [hunk 3]
```diff
@@ -21,7 +21,7 @@ object AvatarHelper {
         listener: AvatarRequestListener
     ) {
         val avatarSize = fragment.resources.getDimensionPixelSize(R.dimen.avatar_sz_login)
-        val avatarUrl = GravatarUtils.gravatarFromEmail(email, avatarSize, STATUS_404)
+        val avatarUrl = email?.let { emailAddressToGravatarUrl(email, avatarSize, DefaultAvatarImage.STATUS_404) }
```
Comment (maxme) [inline WordPressLoginFlow/src/main/java/org/wordpress/android/login/util/AvatarHelper.kt]: Update to use Gravatar library

File: build.gradle [hunk 4]
```diff
@@ -25,7 +20,7 @@ ext {
     androidxLegacySupportV13Version = '1.0.0'
     androidxMediaVersion = '1.2.1'
     androidxVectorDrawableAnimatedVersion = '1.1.0'
-    daggerVersion = '2.42'
+    daggerVersion = '2.46.1'
```
Comment (maxme) [inline build.gradle]: If I don't bump version to the same version used in wpandroid, the build fails with cryptic errors

File: settings.gradle [hunk 5]
```diff
@@ -1,5 +1,5 @@
 pluginManagement {
-    gradle.ext.kotlinVersion = '1.6.10'
+    gradle.ext.kotlinVersion = '1.9.10'
```
Comment (maxme) [inline settings.gradle:2]: Updated to the same kotlin version as wpandroid

**Comment by wzieba** [comment]
hi @maxme ! I've allowed myself to push https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/134/commits/eeea07c0f47772dbac50b851d324e0d6d292832e to test and showcase usage of artifact stored on S3.

The version now is `<PR_number>-<commit-hash>` but later we'll adjust to the desired version via Git tag.

**Comment by maxme** [comment]
@wzieba no problem, I'll update it again with the right version when your PR is merged in `trunk` on the other side.

File: WordPressLoginFlow/build.gradle [hunk 6]
```diff
@@ -32,11 +33,19 @@ android {
...
 
-    compileSdkVersion rootProject.compileSdkVersion
+    kotlinOptions {
+        jvmTarget = JavaVersion.VERSION_1_8
+
+    }
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_1_8
+        targetCompatibility = JavaVersion.VERSION_1_8
+    }
```
Comment (hamorillo) [inline WordPressLoginFlow/build.gradle]: ⚠️ We have another compile options in line 25.

File: WordPressLoginFlow/build.gradle [hunk 7]
```diff
@@ -32,11 +33,19 @@ android {
 
     namespace "org.wordpress.android.login"
 
-    compileSdkVersion rootProject.compileSdkVersion
+    kotlinOptions {
+        jvmTarget = JavaVersion.VERSION_1_8
+
```
Comment (hamorillo) [inline WordPressLoginFlow/build.gradle]:
```suggestion

```

**Review by antonis — COMMENTED** [review]
Hey @maxme 👋 
The code changes on this PR look consistent to me but I wasn't able to successfully test the changes with the WPAndroid app.
I tried both with a hash reference (`134-fbddbe084f24fafd0dc299667674a822bdd9626d`) and a [composite build](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/local-builds.gradle-example#L21) but I get a build error since [gravatar-0.2.0](https://github.com/Automattic/Gravatar-SDK-Android/releases/tag/0.2.0) cannot be retrieved 🤔 

```
* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ...
Caused by: org.gradle.internal.resolve.ModuleVersionNotFoundException: Could not find com.gravatar:gravatar:0.2.0.
Searched in the following locations:
  - https://dl.google.com/dl/android/maven2/com/gravatar/gravatar/0.2.0/gravatar-0.2.0.pom
  - https://repo.maven.apache.org/maven2/com/gravatar/gravatar/0.2.0/gravatar-0.2.0.pom
  - file:/Users/antonis/git/WordPress-Android/aars/gravatar-0.2.0.jar
  - file:/Users/antonis/git/WordPress-Android/aars/gravatar.jar
  - https://www.jitpack.io/com/gravatar/gravatar/0.2.0/gravatar-0.2.0.pom
Required by:
    project :WordPress > org.wordpress:login:134-fbddbe084f24fafd0dc299667674a822bdd9626d
``` 
I've also created a PR to verify that this is not related to my local configuration and the CI build failed too: https://github.com/wordpress-mobile/WordPress-Android/pull/20591

**Comment by hamorillo** [comment]
>  I tried both with a hash reference (`134-fbddbe084f24fafd0dc299667674a822bdd9626d`) and a [composite build](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/local-builds.gradle-example#L21) but I get a build error since [gravatar-0.2.0](https://github.com/Automattic/Gravatar-SDK-Android/releases/tag/0.2.0) cannot be retrieved 🤔

I guess you are using the `trunk` branch of WPAndroid. You can test it with [this branch](https://github.com/wordpress-mobile/WordPress-Android/tree/maxime/remove-gravatar-utils-from-wputils-squashed). The problem is that it's not looking for the library in the right place. We need [these changes](https://github.com/wordpress-mobile/WordPress-Android/blob/e8f23bb0655ba0b577ad3fe60de7ec992f9c705e/WordPress/build.gradle#L47) for that.

**Comment by antonis** [comment]
> I guess you are using the trunk branch of WPAndroid. You can test it with [this branch](https://github.com/wordpress-mobile/WordPress-Android/tree/maxime/remove-gravatar-utils-from-wputils-squashed). The problem is that it's not looking for the library in the right place. We need [these changes](https://github.com/wordpress-mobile/WordPress-Android/blob/e8f23bb0655ba0b577ad3fe60de7ec992f9c705e/WordPress/build.gradle#L47) for that.

That's correct. Thank you for pointing me to this branch @hamorillo 🙇 
I'll retest with that. Are there specific changes I should look for or just a sanity of the functionality?

**Comment by hamorillo** [comment]
The changes are basically that we are using the new Gravatar SDK to load the avatars and generate the URLs so, I would say a sanity check paying attention that you can see the avatars as expected everywhere 🙂.

**Comment by maxme** [comment]
I reverted all the build file changes that felt unnecessary. I get a weird compiling error on my machine when I have WPAndroid setup with  WPLogin-Flow and Gravatar-SDK as source dependencies. I can build WPLogin-Flow alone fine though...

**Review by antonis — APPROVED** [review]
Thank you for your work on this @maxme 🙇 
I've retested the PR with the `WordPress-Android` branch [maxime/remove-gravatar-utils-from-wputils-squashed
](https://github.com/wordpress-mobile/WordPress-Android/compare/trunk...maxime/remove-gravatar-utils-from-wputils-squashed) and the head of this PR.
```
wordPressLoginVersion = '134-150a929218a85a5d0a5aa2aec12a344186e30c88'
```
The App worked as expected for me and haven't noticed any change in the Login/Signup flows or the Me tab were the Gravatar is present 🎉 
I'll be glad to test in more detail along the `WordPress-Android` integration PR.

**Review by ThomazFB — APPROVED** [review]
Hey @maxme, from the Woo side, I can confirm the app builds and runs normally with the changes available in this PR. Here's a patch with the changes I did to the Woo app to make it work:

```
Subject: [PATCH] Support WPLogin changes
---
Index: build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/build.gradle b/build.gradle
--- a/build.gradle	(revision 53c64f14f8d9f73f9f7544cc0bac900544ab1f34)
+++ b/build.gradle	(date 1712236709252)
@@ -116,7 +116,7 @@
     hiltJetpackVersion = '1.1.0'
     wordPressUtilsVersion = '3.5.0'
     mediapickerVersion = '0.3.0'
-    wordPressLoginVersion = '1.13.0'
+    wordPressLoginVersion = '134-150a929218a85a5d0a5aa2aec12a344186e30c88'
     aboutAutomatticVersion = '0.0.6'
     automatticTracksVersion = '4.0.2'
     workManagerVersion = '2.7.1'
Index: WooCommerce/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/build.gradle b/WooCommerce/build.gradle
--- a/WooCommerce/build.gradle	(revision 53c64f14f8d9f73f9f7544cc0bac900544ab1f34)
+++ b/WooCommerce/build.gradle	(date 1712239418918)
@@ -54,6 +54,7 @@
             includeGroup "org.wordpress.mediapicker"
             includeGroup "com.automattic"
             includeGroup "com.automattic.tracks"
+            includeGroup "com.gravatar"
         }
     }
     maven {
```

Looks good to me!

**Comment by maxme** [comment]
Cool thanks everyone for testing, I'll merge this one.

What's the process to publish a new WordPress-Login-Flow-Android release?

**Comment by antonis** [comment]
Thank you for your work on this @maxme 🙇 

> What's the process to publish a new WordPress-Login-Flow-Android release?

My understanding is that we just need to [push a tag and create a new release](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/releases). When the CI takes care of the rest.

We also need to merge the changes on the WPAndroid/Woo clients to avoid any breakages.

**Pull request merged**